core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -390,16 +390,9 @@ private void invalidateCachedTxIds() {
     public int getWeight() {
         if (!hasWitnesses())
             return this.messageSize() * 4;
-        try (final ByteArrayOutputStream stream = new ByteArrayOutputStream(255)) { // just a guess at an average tx length
-            bitcoinSerializeToStream(stream, false);
-            final int baseSize = stream.size();
-            stream.reset();
-            bitcoinSerializeToStream(stream, true);
-            final int totalSize = stream.size();
-            return baseSize * 3 + totalSize;
-        } catch (IOException e) {
-            throw new RuntimeException(e); // cannot happen
-        }
+        int baseSize = messageSize(false);
+        int totalSize = messageSize(true);
+        return baseSize * 3 + totalSize;
```

Comment:
With a ternary this could be functional-style with a single `return` statement.

Reply:
Yes, we can do that. I decided to do the simple refactor first, for easy reviewability.